### ISSUE TITLE
[JENKINS-45917] [Jenkins v2.63] Build queue deadlocks

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#22220]
<p>We are experiencing issues with our jenkins server where it will regularly lock the build queue such that no jobs can be scheduled, and no jobs in the queue start running. At this point we are also unable to use the /script endpoint to clear the build queue</p>

<p> </p>

<p>We have been unable to determine the root cause of this issue, and restarting the jenkins server causes it to come up in a locked state as well. We have attempted to remove all the files that we know of that would tell Jenkins to resume jobs after a restart, but have been unable to find the comprehensive list. In particular are the pipeline jobs that come up and try to resume. Below is the list of files that we have tried removing: </p>

<p> </p>

<pre>
/<span class="code-keyword">var</span>/lib/jenkins/queue.xml
/<span class="code-keyword">var</span>/lib/jenkins/queue.xml.bak
/<span class="code-keyword">var</span>/lib/jenkins/org.jenkinsci.plugins.workflow.flow.FlowExecutionList.xml
/<span class="code-keyword">var</span>/lib/jenkins/org.jenkinsci.plugins.workflow.flow.FlowExecutionList.bak
/<span class="code-keyword">var</span>/lib/jenkins/org.jenkinsci.plugins.workflow.support.steps.StageStep.xml 
</pre>
<p> </p>

<p>Other triage steps we have taken: </p>
<ul>
	<li>Increased JENKINS_HANDLER_MAX from 300 to 600 in /etc/sysconfig/jenkins</li>
	<li>Starting in shutdown mode with a clear build queue</li>
	<li>disabling jobs that queue frequently</li>
</ul>

<p>(first 3 edits were on the same day the ticket was created, Tuesday August 1, 2017)</p>

<p>edit: </p>

<p>We bumped the number of executors to 2, and we see jobs queueing and running, but only on the second executor, not the first one. </p>

<p>edit 2: </p>

<p>Added an image from the /monitoring endpoint, the active thread count is the only sign that something has gone wrong. No other metric appears to be affected in a meaningful way</p>

<p>edit 3:</p>

<p>We managed to get a thread dump from the /threadDump endpoint. There 3 different suspect thread dumps: </p>

<pre>
<span class="code-quote">"AtmostOneTaskExecutor[Periodic Jenkins queue maintenance] [#417]"</span> Id=19597 Group=main WAITING on com.google.common.util.concurrent.AbstractFuture$Sync@​488e43ab at sun.misc.Unsafe.park(Native Method) - waiting on com.google.common.util.concurrent.AbstractFuture$Sync@​488e43ab at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175) at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836) at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997) at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304) at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:275) at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:111) at org.jenkinsci.plugins.workflow.cps.CpsStepContext.getThreadGroupSynchronously(CpsStepContext.java:248) at org.jenkinsci.plugins.workflow.cps.CpsStepContext.getThreadSynchronously(CpsStepContext.java:237) at org.jenkinsci.plugins.workflow.cps.CpsStepContext.doGet(CpsStepContext.java:294) at org.jenkinsci.plugins.workflow.support.DefaultStepContext.get(DefaultStepContext.java:61) at org.jenkinsci.plugins.workflow.support.steps.ExecutorStepExecution$PlaceholderTask.getNode(ExecutorStepExecution.java:259) at hudson.plugins.throttleconcurrents.ThrottleQueueTaskDispatcher.categoriesForPipeline(ThrottleQueueTaskDispatcher.java:411) at hudson.plugins.throttleconcurrents.ThrottleQueueTaskDispatcher.canRun(ThrottleQueueTaskDispatcher.java:168) at hudson.model.Queue.isBuildBlocked(Queue.java:1184) at hudson.model.Queue.maintain(Queue.java:1505) at hudson.model.Queue$1.call(Queue.java:320) at hudson.model.Queue$1.call(Queue.java:317) at jenkins.util.AtmostOneTaskExecutor$1.call(AtmostOneTaskExecutor.java:108) at jenkins.util.AtmostOneTaskExecutor$1.call(AtmostOneTaskExecutor.java:98) at jenkins.security.ImpersonatingExecutorService$2.call(ImpersonatingExecutorService.java:71) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110) at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:748) <span class="code-object">Number</span> of locked synchronizers = 1 - java.util.concurrent.locks.ReentrantLock$NonfairSync@​529a7a50</pre>

<pre>
<span class="code-quote">"Computer.threadPoolForRemoting [#80]"</span> Id=19565 Group=main BLOCKED on hudson.model.Queue@​770f2152 owned by <span class="code-quote">"Computer.threadPoolForRemoting [#84]"</span> Id=19573 at com.nirima.jenkins.plugins.docker.strategy.DockerOnceRetentionStrategy$1.run(DockerOnceRetentionStrategy.java:110) - blocked on hudson.model.Queue@​770f2152 at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28) at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:748) <span class="code-object">Number</span> of locked synchronizers = 1 - java.util.concurrent.ThreadPoolExecutor$Worker@​3677ff7e</pre>

<pre>
<span class="code-quote">"Computer.threadPoolForRemoting [#81]"</span> Id=19569 Group=main TIMED_WAITING on java.util.concurrent.SynchronousQueue$TransferStack@​18c5c88 at sun.misc.Unsafe.park(Native Method) - waiting on java.util.concurrent.SynchronousQueue$TransferStack@​18c5c88 at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215) at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:460) at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362) at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:941) at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1066) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:748)
</pre>
<p> </p>

<p>EDIT 4 (08-03-17|August 3): We seem to have stabilized the server.</p>

<p>In our set up, a pipeline build that was interrupted would attempt to resume upon Jenkins restart. However when it does so it uses the exact node name that it had before which in this case is a docker container name with a unique hash that no longer exists. This causes the build to zombie itself and hang.</p>

<p>The only indication on the file system that a pipeline job has not completed is that most recent stage xml file in the workflow directory for a given build does not contain a "<result>" tag. We wrote a python script that we ran on the jenkins master server to look for that and delete any build that did not have a "<result>" tag. We would appreciate any feed back on this approach, the python script is attached below. </p>

<p>The directory for a given job is: </p>

<p>/var/lib/jenkins/jobs/<JOB_NAME>/builds/<BUILD_NUMBER>/workflow</p>

<p> </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mpedreiro.svg" title="mpedreiro's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mpedreiro">mpedreiro</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-45917" target="_blank">[Jenkins v2.63] Build queue deadlocks</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>label(s)</b>: build-queue, deadlock, jenkins, pipeline, pipeline-hangs, threads
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2017-08-24T13:54:05+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We are experiencing issues with our jenkins server where it will regularly lock the build queue such that no jobs can be scheduled, and no jobs in the queue start running. At this point we are also unable to use the /script endpoint to clear the build queue</p>

<p> </p>

<p>We have been unable to determine the root cause of this issue, and restarting the jenkins server causes it to come up in a locked state as well. We have attempted to remove all the files that we know of that would tell Jenkins to resume jobs after a restart, but have been unable to find the comprehensive list. In particular are the pipeline jobs that come up and try to resume. Below is the list of files that we have tried removing: </p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">/<span class="code-keyword">var</span>/lib/jenkins/queue.xml
/<span class="code-keyword">var</span>/lib/jenkins/queue.xml.bak
/<span class="code-keyword">var</span>/lib/jenkins/org.jenkinsci.plugins.workflow.flow.FlowExecutionList.xml
/<span class="code-keyword">var</span>/lib/jenkins/org.jenkinsci.plugins.workflow.flow.FlowExecutionList.bak
/<span class="code-keyword">var</span>/lib/jenkins/org.jenkinsci.plugins.workflow.support.steps.StageStep.xml 
</pre>
</div></div>
<p> </p>

<p>Other triage steps we have taken: </p>
<ul>
	<li>Increased JENKINS_HANDLER_MAX from 300 to 600 in /etc/sysconfig/jenkins</li>
	<li>Starting in shutdown mode with a clear build queue</li>
	<li>disabling jobs that queue frequently</li>
</ul>

<p>(first 3 edits were on the same day the ticket was created, Tuesday August 1, 2017)</p>

<p>edit: </p>

<p>We bumped the number of executors to 2, and we see jobs queueing and running, but only on the second executor, not the first one. </p>

<p>edit 2: </p>

<p>Added an image from the /monitoring endpoint, the active thread count is the only sign that something has gone wrong. No other metric appears to be affected in a meaningful way</p>

<p>edit 3:</p>

<p>We managed to get a thread dump from the /threadDump endpoint. There 3 different suspect thread dumps: </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-quote">"AtmostOneTaskExecutor[Periodic Jenkins queue maintenance] [#417]"</span> Id=19597 Group=main WAITING on com.google.common.util.concurrent.AbstractFuture$Sync@488e43ab at sun.misc.Unsafe.park(Native Method) - waiting on com.google.common.util.concurrent.AbstractFuture$Sync@488e43ab at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175) at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836) at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997) at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304) at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:275) at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:111) at org.jenkinsci.plugins.workflow.cps.CpsStepContext.getThreadGroupSynchronously(CpsStepContext.java:248) at org.jenkinsci.plugins.workflow.cps.CpsStepContext.getThreadSynchronously(CpsStepContext.java:237) at org.jenkinsci.plugins.workflow.cps.CpsStepContext.doGet(CpsStepContext.java:294) at org.jenkinsci.plugins.workflow.support.DefaultStepContext.get(DefaultStepContext.java:61) at org.jenkinsci.plugins.workflow.support.steps.ExecutorStepExecution$PlaceholderTask.getNode(ExecutorStepExecution.java:259) at hudson.plugins.throttleconcurrents.ThrottleQueueTaskDispatcher.categoriesForPipeline(ThrottleQueueTaskDispatcher.java:411) at hudson.plugins.throttleconcurrents.ThrottleQueueTaskDispatcher.canRun(ThrottleQueueTaskDispatcher.java:168) at hudson.model.Queue.isBuildBlocked(Queue.java:1184) at hudson.model.Queue.maintain(Queue.java:1505) at hudson.model.Queue$1.call(Queue.java:320) at hudson.model.Queue$1.call(Queue.java:317) at jenkins.util.AtmostOneTaskExecutor$1.call(AtmostOneTaskExecutor.java:108) at jenkins.util.AtmostOneTaskExecutor$1.call(AtmostOneTaskExecutor.java:98) at jenkins.security.ImpersonatingExecutorService$2.call(ImpersonatingExecutorService.java:71) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:110) at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:748) <span class="code-object">Number</span> of locked synchronizers = 1 - java.util.concurrent.locks.ReentrantLock$NonfairSync@529a7a50</pre>
</div></div>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-quote">"Computer.threadPoolForRemoting [#80]"</span> Id=19565 Group=main BLOCKED on hudson.model.Queue@770f2152 owned by <span class="code-quote">"Computer.threadPoolForRemoting [#84]"</span> Id=19573 at com.nirima.jenkins.plugins.docker.strategy.DockerOnceRetentionStrategy$1.run(DockerOnceRetentionStrategy.java:110) - blocked on hudson.model.Queue@770f2152 at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28) at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:748) <span class="code-object">Number</span> of locked synchronizers = 1 - java.util.concurrent.ThreadPoolExecutor$Worker@3677ff7e</pre>
</div></div>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-quote">"Computer.threadPoolForRemoting [#81]"</span> Id=19569 Group=main TIMED_WAITING on java.util.concurrent.SynchronousQueue$TransferStack@18c5c88 at sun.misc.Unsafe.park(Native Method) - waiting on java.util.concurrent.SynchronousQueue$TransferStack@18c5c88 at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215) at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:460) at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362) at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:941) at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1066) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1127) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) at java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:748)
</pre>
</div></div>
<p> </p>

<p>EDIT 4 (08-03-17|August 3): We seem to have stabilized the server.</p>

<p>In our set up, a pipeline build that was interrupted would attempt to resume upon Jenkins restart. However when it does so it uses the exact node name that it had before which in this case is a docker container name with a unique hash that no longer exists. This causes the build to zombie itself and hang.</p>

<p>The only indication on the file system that a pipeline job has not completed is that most recent stage xml file in the workflow directory for a given build does not contain a "&lt;result&gt;" tag. We wrote a python script that we ran on the jenkins master server to look for that and delete any build that did not have a "&lt;result&gt;" tag. We would appreciate any feed back on this approach, the python script is attached below. </p>

<p>The directory for a given job is: </p>

<p>/var/lib/jenkins/jobs/&lt;JOB_NAME&gt;/builds/&lt;BUILD_NUMBER&gt;/workflow</p>

<p> </p></pre>
</details>
</details>
<details><summary><i>2 attachments</i></summary>

- [active thead count.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/39114/active%20thead%20count.png)
  > ![active thead count.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/39114/active%20thead%20count.png)
- [clean_zombie_jobs.py](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/39140/clean_zombie_jobs.py)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/integer.png" title="integer's avatar" /> integer:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-45917?focusedId=309025&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-309025">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>logic come from <a href="https://github.com/jenkinsci/durable-task-plugin/blob/master/src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/durable-task-plugin/blob/master/src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java</a> i guess non-pipeline jobs has no issues. </p>
</pre>
</details>

</ul>
</details>

<p>logic come from <a href="https://github.com/jenkinsci/durable-task-plugin/blob/master/src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/durable-task-plugin/blob/master/src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java</a> i guess non-pipeline jobs has no issues. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/maxfields2000.jpg" title="maxfields2000's avatar" /> maxfields2000:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-45917?focusedId=309045&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-309045">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I was theorizing that the Docker/Jclouds Plugins need to inspect the build queue to decide if they need to provision a new node.  I don't have heap dumps/thread dumps from our Jenkins server that doesn't use the Docker/Jclouds pplugins but we have seen the build queue hang on that server after restart as well. In fact we first saw the behavior there.  Your findings however are hopeful that potentially there's a way to fix this in the future.</p>
</pre>
</details>

</ul>
</details>

<p>I was theorizing that the Docker/Jclouds Plugins need to inspect the build queue to decide if they need to provision a new node.  I don't have heap dumps/thread dumps from our Jenkins server that doesn't use the Docker/Jclouds pplugins but we have seen the build queue hang on that server after restart as well. In fact we first saw the behavior there.  Your findings however are hopeful that potentially there's a way to fix this in the future.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/maxfields2000.jpg" title="maxfields2000's avatar" /> maxfields2000:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-45917?focusedId=309047&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-309047">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=integer" class="user-hover" rel="integer">integer</a> Would switching to a different Retention strategy avoid this? I have to look at the Yet Another Docker Plugin to see what options exist that might emulate the behavior of provisioning and throwing away the container when done. Or do all the retention strategies have a lock on the build queue?  (I know this doesn't appear specific to the Docker plugins as tghe "Once Retention Strategy" is not just a Docker thing).</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=integer" class="user-hover" rel="integer">integer</a> Would switching to a different Retention strategy avoid this? I have to look at the Yet Another Docker Plugin to see what options exist that might emulate the behavior of provisioning and throwing away the container when done. Or do all the retention strategies have a lock on the build queue?  (I know this doesn't appear specific to the Docker plugins as tghe "Once Retention Strategy" is not just a Docker thing).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/integer.png" title="integer's avatar" /> integer:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-45917?focusedId=309068&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-309068">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I also see ghprb plugin with some weird locks, maybe try disable it. </p>
</pre>
</details>

</ul>
</details>

<p>I also see ghprb plugin with some weird locks, maybe try disable it. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-45917?focusedId=310003&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310003">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>CC <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=abayer" class="user-hover" rel="abayer">abayer</a> and <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=svanoort" class="user-hover" rel="svanoort">svanoort</a> ^^^</p>
</pre>
</details>

</ul>
</details>

<p>CC <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=abayer" class="user-hover" rel="abayer">abayer</a> and <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=svanoort" class="user-hover" rel="svanoort">svanoort</a> ^^^</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>svanoort:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-45917?focusedId=311579&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-311579">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> yes, it should - albeit after a 5 minute timeout before the build aborts when it can't resume on the same node (safety measure).  <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mpedreiro" class="user-hover" rel="mpedreiro">mpedreiro</a> can you try installing the latest version of durable task step plugin and let us know if that doesn't settle it out (bear in mind that there is a delay before the build is killed when it cannot resume cleanly). </p>

<p>Thanks!</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> yes, it should - albeit after a 5 minute timeout before the build aborts when it can't resume on the same node (safety measure).  <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mpedreiro" class="user-hover" rel="mpedreiro">mpedreiro</a> can you try installing the latest version of durable task step plugin and let us know if that doesn't settle it out (bear in mind that there is a delay before the build is killed when it cannot resume cleanly). </p>

<p>Thanks!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>svanoort:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-45917?focusedId=311581&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-311581">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Closing as duplicate of the linked issue, since the fix there should cover this as well.</p>
</pre>
</details>

</ul>
</details>

<p>Closing as duplicate of the linked issue, since the fix there should cover this as well.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `duplicates` from Jira: <a href="https://github.com/jenkinsci/jenkins/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-36013%22">JENKINS-36013</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/jenkins/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-33761%22">JENKINS-33761</a>]</i>
